File: src/main/java/com/cottonlesergal/ucontrolbot/models/dto/UserDTO.java
package com.cottonlesergal.ucontrolbot.models.dto;

import net.dv8tion.jda.api.entities.User;

public class UserDTO extends BaseDTO {
    // Constants for user properties
    public static final String ID = "id";
    public static final String USERNAME = "username";
    public static final String DISPLAY_NAME = "displayName";
    public static final String AVATAR = "avatarUrl";
    public static final String DISCRIMINATOR = "discriminator";
    public static final String IS_BOT = "isBot";
    public static final String STATUS = "status";

    public UserDTO() {
        // Default constructor
    }

    public UserDTO(User jdaUser) {
        if (jdaUser != null) {
            set(ID, jdaUser.getId());
            set(USERNAME, jdaUser.getName());
            set(AVATAR, jdaUser.getEffectiveAvatarUrl());
            set(DISCRIMINATOR, jdaUser.getDiscriminator());
            set(IS_BOT, jdaUser.isBot());
        }
    }

    // Add user-specific methods
    public String getId() {
        return (String) get(ID);
    }

    public String getUsername() {
        return (String) get(USERNAME);
    }

    public String getDisplayName() {
        return (String) get(DISPLAY_NAME);
    }

    public String getAvatarUrl() {
        return (String) get(AVATAR);
    }

    public UserDTO fetchInfo(String userId) {
        // Implement fetch logic
        return this;
    }
}

File: src/main/resources/static/js/dto/UserDTO.js
/**
 * User DTO for Discord user entities
 * @extends DiscordDTO
 */
class UserDTO extends DiscordDTO {
    // Property keys
    static ID = 'id';
    static USERNAME = 'username';
    static DISPLAY_NAME = 'displayName';
    static AVATAR = 'avatarUrl';
    static DISCRIMINATOR = 'discriminator';
    static IS_BOT = 'isBot';
    static STATUS = 'status';

    /**
     * Create a new UserDTO
     * @param {Object} data - Initial data
     */
    constructor(data = {}) {
        super(data);
    }

    /**
     * Get user ID
     * @returns {string} - User ID
     */
    getId() {
        return this.get(UserDTO.ID);
    }

    /**
     * Get username
     * @returns {string} - Username
     */
    getUsername() {
        return this.get(UserDTO.USERNAME);
    }

    /**
     * Get display name
     * @returns {string} - Display name
     */
    getDisplayName() {
        return this.get(UserDTO.DISPLAY_NAME);
    }

    /**
     * Get avatar URL
     * @returns {string} - Avatar URL
     */
    getAvatarUrl() {
        return this.get(UserDTO.AVATAR);
    }

    /**
     * Fetch user info from API
     * @param {string} userId - User ID
     * @returns {Promise<UserDTO>} - This UserDTO with updated data
     */
    async fetchInfo(userId) {
        try {
            const response = await fetch(`/api/users/${userId}`);
            if (response.ok) {
                const data = await response.json();
                this.fromJSON(data);
            }
        } catch (error) {
            console.error('Failed to fetch user:', error);
        }
        return this;
    }
}

// Make available globally
window.UserDTO = UserDTO;